plaidbench.py

```diff
@@ -111,125 +216,131 @@ def main():
-            output.contents = y
-            print('Warmup')
- 
-            for i in range(32 // batch_size + 1):
-                y = model.predict(x=x_train, batch_size=batch_size)
-            # Now start the clock and run 100 batches
-            print('Doing the main timing')
-
-            for i in range(examples // batch_size):
-                stop_watch.start()
-                y = model.predict(x=x_train, batch_size=batch_size)
-                stop_watch.stop()
-                time.sleep(.025 * random.random())
+            data['plaid'] = None
 
-        stop_watch.stop()
-        compile_stop_watch.stop()
-        execution_duration = stop_watch.elapsed() / examples
-        compile_duration = compile_stop_watch.elapsed()
-        data['execution_duration'] = execution_duration
-        data['compile_duration'] = compile_duration
-        printf('Example finished, elapsed: {} (compile), {} (execution, per example)'.format(
-            compile_duration, execution_duration))
-        data['precision'] = output.precision
-    except Exception as ex:
-        printf(ex)
-        data['exception'] = str(ex)
-        exit_status = -1
-        if args.print_stacktraces:
-            raise
-        printf('Set --print-stacktraces to see the entire traceback')
-    finally:
+        data['example_size'] = examples
+        data['train'] = args.train
+        data['blanket_run'] = True
+        outputs['run_configuration'] = data.copy()
+    else:
+        networks.append(args.module)
+
+    for network in networks:
+        printf("\nCurrent network being run : " + network)  
+        args.module = network
+        network_data = {}
+
+        # Run network
+        try:
+            # Setup
+            x_train, y_train = setup(args.train, epoch_size, batch_size)
+
+            # Loading the model
+            module, x_train, model = load_model(args.module, x_train)
+
+            # training run
+            if args.train:
+                value_check(examples, epochs, batch_size)
+                train(x_train, y_train, epoch_size, model, batch_size, 
+                    compile_stop_watch, epochs, stop_watch, output, network)
+            # inference run
+            else:
+                inference(args.module, model, batch_size, compile_stop_watch, 
+                    output, x_train, examples, stop_watch)
+
+            # Record stopwatch times
+            execution_duration = stop_watch.elapsed()
+            compile_duration = compile_stop_watch.elapsed()
+            
+            # Record data
+            if args.blanket_run:
+                network_data['average_example_duration'] = execution_duration / examples
+
+            network_data['execution_duration'] = execution_duration
+            network_data['compile_duration'] = compile_duration
+            network_data['precision'] = output.precision
+
+            # Print statement
+            printf('Example finished, elapsed: {} (compile), {} (execution)'.format(
```

Comment:
In general, it's probably better to use keyword arguments over positional for format strings.